compositor_render/src/renderer/scene.rs

```diff
@@ -125,6 +131,35 @@ impl RenderNode {
             RenderNode::Builtin(node) => node.resolution_from_spec(),
         }
     }
+
+    fn should_fallback(&self, sources: &[(&NodeId, &NodeTexture)]) -> bool {
+        match self.fallback_strategy() {
+            FallbackStrategy::NeverFallback => false,
+            FallbackStrategy::FallbackIfAllInputsMissing => sources
+                .iter()
+                .all(|(_, node_texture)| node_texture.is_empty()),
+            FallbackStrategy::FallbackIfAnyInputsMissing => sources
+                .iter()
+                .any(|(_, node_texture)| node_texture.is_empty()),
+            FallbackStrategy::FallbackIfOnlyInputMissing => {
+                sources.len() == 1
+                    && sources
+                        .iter()
+                        .all(|(_, node_texture)| node_texture.is_empty())
+            }
+        }
+    }
```

Comment:
hsouldn't this be defined in `impl FallbackStrategy`? it seems weird to have it defined in here

Reply:
We can't move it to impl FallbackStrategy, since it has to be located in `compositor_common`, which should be independent of `compositor_renderer`.
We would need to define some entity that would represent the state of input texture (whether it's empty), map sources to it and then run that method it `compositor_common`. IMO it's not worth it.

Reply:
I can't see any upsides of creating this trait. It's so specific, that it would be hard to use it anywhere else (we would need to have access to `sources`, which are very specific for node).

I can implement mapping, but not sure whether it's useful in any way.
What do you think @wkozyra95 ?

Reply:
as I said, it's up to you. I like making things like this reusable in the future, but it's fine to leave it as-is